src/System.Runtime/tests/System/String.Split.cs

```diff
@@ -0,0 +1,515 @@
+    {
+        return value.Split(new[] { separator }, StringSplitOptions.None);
+    }
+
+    public static string[] Split(this string value, string separator, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, options);
+    }
+
+    public static string[] Split(this string value, string separator, int count, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, count, options);
+    }
+}
+
+public static class StringSplitTests
+{
+    [Fact]
+    public static void TestSplitInvalidCount()
+    {
+        const string value = "a,b";
+        const int count = -1;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(',', count, options));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { ',' }, count));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { ',' }, count, options));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(",", count, options));
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitInvalidOptions()
+    {
+        const string value = "a,b";
+        const int count = int.MaxValue;
+        const StringSplitOptions optionsTooLow = StringSplitOptions.None - 1;
+        const StringSplitOptions optionsTooHigh = StringSplitOptions.RemoveEmptyEntries + 1;
+
+        Assert.Throws<ArgumentException>(() => value.Split(',', optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(',', optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(',', count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(',', count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { ',' }, count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(",", optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(",", optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(",", count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(",", count, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, optionsTooHigh));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, count, optionsTooLow));
+        Assert.Throws<ArgumentException>(() => value.Split(new[] { "," }, count, optionsTooHigh));
+    }
+
```

Comment:
I just finished some Linq tests that also use extensions and they throw a ArgumentException when the source is null. You're current temporary implementation could also be trivially modified to throw a ArgumentException rather than a NullReferenceException, so it's just defensive against someone changing the implementation of the extension methods.